flask/app.py

```diff
@@ -1611,12 +1611,17 @@ def _find_error_handler(self, e):
         """
         exc_class, code = self._get_exc_class_and_code(type(e))
 
+        _order = []
         for name, c in (
             (request.blueprint, code), (None, code),
             (request.blueprint, None), (None, None)
         ):
-            handler_map = self.error_handler_spec.setdefault(name, {}).get(c)
+            if (name, c) in _order:
```

Comment:
The other change further down is fine, but I don't find this one necessary.

Reply:
If I can use orderedset, it wouldn't be so ugly!